File: process_mascots.py
#!/usr/bin/env python3
"""
Process mascot images and remove black backgrounds
Install: pip install Pillow
Run: python process_mascots.py
"""
import sys
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("PIL not found. Install with: pip install Pillow")
    sys.exit(1)

def remove_background_and_save(image_data: bytes, output_path: str, threshold: int = 40):
    """Convert image bytes to PNG with transparent background"""
    try:
        # Open image from bytes
        img = Image.open(Path(output_path).parent / f"temp_{Path(output_path).name}")
        
        # Ensure RGBA
        if img.mode != 'RGBA':
            img = img.convert('RGBA')
        
        data = img.getdata()
        new_data = []
        
        # Make dark pixels transparent
        for item in data:
            r, g, b, a = item if len(item) == 4 else (item[0], item[1], item[2], 255)
            # If very dark (nearly black), make transparent
            if r < threshold and g < threshold and b < threshold:
                new_data.append((r, g, b, 0))
            else:
                new_data.append((r, g, b, 255))
        
        img.putdata(new_data)
        img.save(output_path, 'PNG')
        print(f"✓ Saved: {output_path}")
        return True
    except Exception as e:
        print(f"✗ Error: {e}")
        return False

# Image mapping
images = {
    'rocket_mascot.png': 'Red/orange rocket with mascot (1st image)',
    'css_mascot.png': 'Blue CSS rocket with 2 characters (2nd image)',
    'sql_mascot.png': 'Red SQL character dancing (3rd image)',
    'python_mascot.png': 'Blue/yellow Python character with thumbs up (4th image)'
}

print("=" * 60)
print("KIDS LOADER MASCOT IMAGE SETUP")
print("=" * 60)
print("\nTo complete setup:")
print("\n1. Download the 4 mascot images from the attachments:")
for fname, desc in images.items():
    print(f"   - {fname}: {desc}")

print("\n2. Save them to: public/images/mascots/")
print("\n3. Run this script again to remove black backgrounds")
print("\nNote: Transparent PNG images are required for the animation!")
print("=" * 60 + "\n")

# Create directory
mascots_dir = Path('public/images/mascots')
mascots_dir.mkdir(parents=True, exist_ok=True)
print(f"✓ Directory created: {mascots_dir}")

File: process-mascots.mjs
#!/usr/bin/env node

/**
 * Process mascot images and remove black backgrounds
 * Install: npm install sharp
 * Run: node process-mascots.mjs
 */

import fs from 'fs';
import path from 'path';

const mascotDir = 'public/images/mascots';

// Create directory if it doesn't exist
if (!fs.existsSync(mascotDir)) {
  fs.mkdirSync(mascotDir, { recursive: true });
  console.log(`✓ Created directory: ${mascotDir}`);
}

console.log('\n' + '='.repeat(60));
console.log('KIDS LOADER MASCOT SETUP');
console.log('='.repeat(60));
console.log('\nTo set up mascot images:');
console.log('\n1. Install image processing library:');
console.log('   npm install sharp');
console.log('\n2. Replace placeholder images with actual mascots:');
console.log('   (Transparent PNG format required)');

const mascots = [
  { name: 'rocket_mascot.png', size: '80×80', source: '1st image (red rocket)' },
  { name: 'css_mascot.png', size: '120×140', source: '2nd image (blue CSS)' },
  { name: 'sql_mascot.png', size: '120×140', source: '3rd image (red SQL)' },
  { name: 'python_mascot.png', size: '120×140', source: '4th image (blue/yellow)' },
];

mascots.forEach(m => {
  const filepath = path.join(mascotDir, m.name);
  console.log(`\n   📦 ${m.name}`);
  console.log(`      Size: ${m.size}, From: ${m.source}`);
});

console.log('\n' + '='.repeat(60) + '\n');

// Create minimal PNG metadata files as placeholders
mascots.forEach(m => {
  const filepath = path.join(mascotDir, m.name);
  if (!fs.existsSync(filepath)) {
    // Create empty file as placeholder
    fs.writeFileSync(filepath, Buffer.alloc(0));
    console.log(`✓ Placeholder: ${filepath}`);
  }
});

console.log('\n✓ Setup complete. Replace placeholder files with actual images.');
